### DISCUSSION TITLE
PSADT V4 option to delay instead of defer

### DISCUSSION THREAD

**Discussion by Pollewops** [PSAppDeployToolkit/PSAppDeployToolkit#1087] [Q&A] [answered]
With Intune, defer does not work.
Please add instead of it a 'delay' option which is configurable with e.g. 1, 2 hour or 4 hours.
After that the installation will continue and install.
I created it myself in v3 and works perfect. Would be nice to have it in v4 as well.

**Comment by mjr4077au** [comment]
The problem with such an option is that its wholly dependent on setting the app's installation timeout in Intune beyond the default 60 minutes. You're also stalling the Intune Management Extension as well as it won't move on to another app while this one's still technically running.

I'm not saying no, but it just requires careful consideration. Our v4 release is also on feature lock as well, we're only doing bug fixes and other minor remediations in the lead up to the release next week. If something like this was to be considered and implemented, it'd have to wait until v4.1.

**Reply by Pollewops** [reply]
At least consideration seems fine with me. Else i create my own extension for it some day.

Tnx

**Reply by mjr4077au** [reply]
Extensions for v4 are going to be better than ever being a module-based architecture. You'd be able to publish your extension in the PSGallery for anyone to download via `Install-Module`, etc.

**Reply by Pollewops** [reply]
Wow creat. Can't wait to test and create

**Comment by mjr4077au** [comment, marked as answer]
@Pollewops a solution for this issue has been submitted by a community member in https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/pull/1254. The PR has been merged into our develop branch for 4.1.0.

I'd encourage you to download the latest develop build artifact from the Actions page and test out the newly implemented feature. If there's any issues, raise it as a new issue so it can be further discussed.

![image](https://github.com/user-attachments/assets/627723e6-ae96-49c1-ad2b-1ddad0b8bc15)
